### PR TITLE
OCPCLOUD-1770: IBMCloud: Use public machine phases 

### PR DESCRIPTION
**Pull request by cjschaef**
Use the public consts for machine phases for consistency.

Related: https://issues.redhat.com/browse/OCPCLOUD-1770

Requires: https://github.com/openshift/api/pull/1358

### CODE REVIEW

**Comment by cjschaef** [comment]
This requires changes to the `openshift/api` to be merged first, so we can properly update the go modules, rather than overriding the api module to my fork.
/hold

File: go.mod [hunk 1]
```diff
@@ -122,9 +123,11 @@ require (
...
 	k8s.io/kubectl v0.25.1 // indirect
-	k8s.io/utils v0.0.0-20220823124924-e9cbc92d1a73 // indirect
 	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
 	sigs.k8s.io/kustomize/api v0.12.1 // indirect
 	sigs.k8s.io/kustomize/kyaml v0.13.9 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
+
+// TODO(cjschaef): Remove before merging
+replace github.com/openshift/api => github.com/cjschaef/api v0.0.0-20221117200200-a44779da707f
```
Comment (JoelSpeed) [inline go.mod:133]: This can be fixed now right?

Comment (cjschaef) [reply go.mod:133]: Yep, I'll be working on updating MAPI and MAO with the proper API go mod and perform another round of testing to confirm

**Comment by cjschaef** [comment]
https://github.com/openshift/api/pull/1358 has merged.
I will rebase and update go mod

File: go.mod (hunk 1 above)
Comment (cjschaef) [reply go.mod]: Tested rebase with custom image. PR has been updated using latest API module.

**Review by elmiko — COMMENTED** [review]
very cool, thanks @cjschaef 
/lgtm

**Comment by cjschaef** [comment]
`e2e-ibmcloud` looks to have a Conformance test failure
```
 Failing tests:
[sig-arch] Managed cluster should ensure platform components have system-* priority class associated [Suite:openshift/conformance/parallel] 
```
```
Dec  1 19:35:51.469: FAIL: 
3 pods found with invalid priority class (should be openshift-user-critical or begin with system-):
openshift-multus/cni-sysctl-allowlist-ds-m9hvp (currently "")
openshift-multus/cni-sysctl-allowlist-ds-vxkfp (currently "")
openshift-multus/cni-sysctl-allowlist-ds-zqsnv (currently "")
```
Unrelated to IBM MAPI.
/retest

**Comment by cjschaef** [comment]
Rebase completed after local testing. Linting looks good, should be ready for review now.
/unhold

**Pull request merged**